lib/builder.js

```diff
@@ -170,11 +173,8 @@ module.exports = class Builder extends EventEmitter {
   makeNodeWrapper(node, _stack) {
     if (_stack == null) _stack = [];
 
-    // Dedupe nodes reachable through multiple paths
-    for (let i = 0; i < this.nodeWrappers.length; i++) {
-      if (this.nodeWrappers[i].originalNode === node) {
-        return this.nodeWrappers[i];
-      }
+    if (this._nodeWrappers.has(node)) {
```

Comment:
Just curious if you prefer this setup (using has + get) over using get directly and guarding for undefined results? 

e.g. this:

```js
let wrapper = this._nodeWrappers.get(node);

if (wrapper !== undefined) {
  return wrapper;
}
```

Since we never expect to _actually_ store `undefined` ^ would "work". I'm not questioning or asking you to change this, I'm just curious about your preference in general...